### PR TITLE
Bundle flvjs and shaka and move all libs to one folder

### PR DESCRIPTION
**Pull request by dkanada**

### CODE REVIEW

File: src/bundle.js [hunk 1]
```diff
@@ -21,6 +25,10 @@ _define("hlsjs", function() { return hlsjs; });
 var howler = require("howler");
 _define("howler", function() { return howler; });
 
+// shaka
+var shaka = require("shaka-player");
+_define("shaka", function() { return shaka });
```
Comment (JustAMan) [inline src/bundle.js]:
```suggestion
_define("shaka", function() { return shaka; });
```

so that style is consistent 😄

File: src/components/viewContainer.js [hunk 2]
```diff
@@ -44,11 +44,11 @@ define(["browser", "dom", "layoutManager", "css!components/viewManager/viewConta
 
 
             if (isPluginpage) {
-                require(["legacy/dashboard"]);
+                require(["legacyDashboard"]);
```
Comment (JustAMan) [inline src/components/viewContainer.js]: could you explain this?

File: src/controllers/user/menu.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ define(["apphost", "connectionManager", "listViewStyle", "emby-button"], functio
             page.querySelector(".lnkMyProfile").setAttribute("href", "myprofile.html?userId=" + userId);
             page.querySelector(".lnkDisplayPreferences").setAttribute("href", "mypreferencesdisplay.html?userId=" + userId);
             page.querySelector(".lnkHomePreferences").setAttribute("href", "mypreferenceshome.html?userId=" + userId);
-            page.querySelector(".lnkLanguagePreferences").setAttribute("href", "mypreferenceslanguages.html?userId=" + userId);
+            page.querySelector(".lnkPlaybackPreferences").setAttribute("href", "mypreferencesplayback.html?userId=" + userId);
```
Comment (JustAMan) [inline src/controllers/user/menu.js]: eh? why?

File: src/scripts/site.js [hunk 4]
```diff
@@ -309,6 +311,10 @@ var AppInfo = {};
         return "bower_components";
     }
 
+    function getComponentsPath() {
```
Comment (JustAMan) [inline src/scripts/site.js]: why is this needed?

File: src/scripts/site.js [hunk 5]
```diff
@@ -381,14 +387,13 @@ var AppInfo = {};
...
             });
         } else {
-            define("ResizeObserver", ["thirdparty/resize-observer-polyfill/ResizeObserver"], returnFirstDependency);
+            define("ResizeObserver", [getBowerPath() + "/resize-observer-polyfill/ResizeObserver"], returnFirstDependency);
         }
     }
 
     function initRequireWithBrowser(browser) {
         var bowerPath = getBowerPath();
-        var apiClientBowerPath = bowerPath + "/apiclient";
```
Comment (JustAMan) [inline src/scripts/site.js]: why remove that and copy-paste `/apiclient` all over the place?

File: src/scripts/site.js [hunk 6]
```diff
@@ -706,35 +680,87 @@ var AppInfo = {};
...
+        define("appFooter-shared", ["appFooter"], createSharedAppFooter);
+
+        // TODO pull apiclient out of this repository
+        define('events', [bowerPath + "/apiclient/events"], returnFirstDependency);
+        define('credentialprovider', [bowerPath + "/apiclient/credentialprovider"], returnFirstDependency);
+        define('connectionManagerFactory', [bowerPath + "/apiclient/connectionmanager"], returnFirstDependency);
+        define('appStorage', [bowerPath + "/apiclient/appStorage"], returnFirstDependency);
+        define("serversync", [bowerPath + "/apiclient/sync/serversync"], returnFirstDependency);
+        define("multiserversync", [bowerPath + "/apiclient/sync/multiserversync"], returnFirstDependency);
+        define("mediasync", [bowerPath + "/apiclient/sync/mediasync"], returnFirstDependency);
```
Comment (JustAMan) [inline src/scripts/site.js]: are these three `-sync` things needed?

File: src/components/viewContainer.js (hunk 2 above)
Comment (dkanada) [reply src/components/viewContainer.js]: I'm just consolidating the legacy script locations in `site.js` so we have a central location for items we need to remove eventually.

File: src/controllers/user/menu.js (hunk 3 above)
Comment (dkanada) [reply src/controllers/user/menu.js]: This is the page that also controls other aspects of media playback like bitrate and skip length, among other things.

File: src/scripts/site.js (hunk 4 above)
Comment (dkanada) [reply src/scripts/site.js]: I was just matching the `getBowerPath()` method above, we might want to eventually remove this but right now the path is set in multiple locations.

File: src/scripts/site.js (hunk 5 above)
Comment (dkanada) [reply src/scripts/site.js]: I did this because the goal is to eventually remove `bowerPath` entirely and pull the dependencies in during the build, and consolidating all of the libraries to remove under `bowerPath` helps with searching for now.

File: src/scripts/site.js (hunk 6 above)
Comment (dkanada) [reply src/scripts/site.js]: I tried removing some of these but was getting a ton of errors, I'll look into it more once we bundle the API client during build.
